bayesian/utility/v2/component_analyses.py

```diff
@@ -0,0 +1,153 @@
+# Copyright © 2020 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author: Deepak Sharma <deepshar@redhat.com>
+#
+"""Component Analyses Utility Stand."""
+
+import logging
+import os
+import re
+from collections import namedtuple
+from typing import Dict, Set, List, Tuple
+from flask import g
+from bayesian.utility.db_gateway import GraphAnalyses
+from bayesian.utils import check_for_accepted_ecosystem, \
+    server_create_analysis, server_create_component_bookkeeping
+from f8a_worker.utils import MavenCoordinates, case_sensitivity_transform
+from werkzeug.exceptions import BadRequest
+
+logger = logging.getLogger(__name__)
+Package = namedtuple("Package", ["name", "version"])
+
+
+def validate_version(version: str) -> bool:
+    """Version should not contain special Characters."""
+    if re.findall('[!@#$%^&*()]', version):
+        return False
+    return True
+
+
+def normlize_packages(packages: List[Dict]) -> List[Package]:
+    """Normalise Packages into hashable."""
+    return [Package(p['name'], p['version']) for p in packages]
+
+
+def unknown_package_flow(ecosystem: str, unknown_pkgs: Set[namedtuple]) -> bool:
+    """Unknown Package flow. Trigger bayesianApiFlow."""
+    api_flow: bool = os.environ.get("INVOKE_API_WORKERS", "") == "1"
+    for pkg in unknown_pkgs:
+        server_create_analysis(ecosystem, pkg.name, pkg.version, user_profile=g.decoded_token,
+                               api_flow=api_flow, force=False, force_graph_sync=True)
```

Comment:
This is a sync call & it would affect the response time. Unknown flow should be executed in the background.

Reply:
That's a really good Suggestion. 
Actually, for Async call, we have to trigger `Event Loop`. We need to figure out whats overhead of that. Need to do some real-time benchmarking to figure out actual Gain. 

This is something that needs R&D, so I will be putting this into Next Iteration. Thanks

Reply:
You can consider using Futures instead of Async io similar to backbone request.

Reply:
Can you add your suggested changes here.

Reply:
Thanks, but As stated earlier, I will not be consideting this recommendation now. 

Please let me know if you have any other recommendations?